packages/payload/src/collections/config/types.ts

```diff
@@ -597,3 +612,18 @@ export type TypeWithTimestamps = {
   id: number | string
   updatedAt: string
 }
+
+export type CompoundIndex = {
```

Comment:
This is great. Having the object shape will let us extend in the future too to support DB specific features like mongodb `partialFilterExpression` and add `dbName` for the postgres char length.